packages/web/components/earn/allocation/index.tsx

```diff
@@ -0,0 +1,143 @@
+import classNames from "classnames";
+import { useMemo } from "react";
+import { useRef } from "react";
+import { useScroll } from "react-use";
+
+import { useTranslation } from "~/hooks";
+import useScrollMeasure from "~/hooks/use-scroll-measure";
+
+const mockTokenRows = [
+  {
+    image: "",
+    name: "ATOM",
+    perc: 23.87,
+  },
+  {
+    image: "",
+    name: "OSMO",
+    perc: 22.87,
+  },
+  {
+    image: "",
+    name: "MARS",
+    perc: 18.87,
+  },
+  {
+    image: "",
+    name: "ETH",
+    perc: 13.87,
+  },
+  {
+    image: "",
+    name: "ATOM",
+    perc: 23.87,
+  },
+  {
+    image: "",
+    name: "OSMO",
+    perc: 22.87,
+  },
+  {
+    image: "",
+    name: "MARS",
+    perc: 18.87,
+  },
+  {
+    image: "",
+    name: "ETH",
+    perc: 13.87,
+  },
+  {
+    image: "",
+    name: "ATOM",
+    perc: 23.87,
+  },
+  {
+    image: "",
+    name: "OSMO",
+    perc: 22.87,
+  },
+  {
+    image: "",
+    name: "MARS",
+    perc: 18.87,
+  },
+  {
+    image: "",
+    name: "ETH",
+    perc: 13.87,
+  },
+];
+
+export const EarnAllocation = () => {
+  const containerRef = useRef(null);
+  const { measure } = useScrollMeasure(containerRef);
+  const { y } = useScroll(containerRef);
+  const { t } = useTranslation();
+
+  const hasReachedBottom = useMemo(
+    () => y + measure.offsetHeight < measure.scrollHeight - 10,
+    [measure.offsetHeight, measure.scrollHeight, y]
+  );
+
+  return (
+    <div className="flex flex-col gap-7 1.5xl:flex-1">
+      <div className="flex items-center gap-3.5">
+        <h5 className="text-lg font-semibold leading-normal text-osmoverse-100 1.5xl:hidden">
+          {t("earnPage.allocation")}
+        </h5>
+        <p className="text-sm font-semibold text-wosmongton-300">5 tokens</p>
+      </div>
+      <div className="relative flex flex-col justify-between gap-12">
+        <div className="flex gap-4">
+          <button className="text-sm font-semibold text-osmoverse-100">
```

Comment:
I think we have existing `<Button />` component. Are we able to use?

Reply:
These buttons are going to be replaced by `Tab` instances, as I already did in the Position / Allocation layout on smaller devices

![Screenshot 2023-12-14 alle 11 01 42](https://github.com/osmosis-labs/osmosis-frontend/assets/31211801/0a4a7993-4e5a-45c8-af8e-e4672d223bb5)